tests/unit/test_armada.py

```diff
@@ -1,71 +1,82 @@
+from concurrent import futures
+
 import grpc
 import pytest
-from armada_client.armada import submit_pb2
+
+from armada_client.armada.submit_pb2 import IngressConfig, ServiceConfig, ServiceType
 from armada_client.client import ArmadaClient
 from armada_client.k8s.io.api.core.v1 import generated_pb2 as core_v1
-from armada_jupyter.armada import (
-    create_armada_request,
-    create_jupyter_ingress,
-    create_jupyter_service,
+from armada_client.k8s.io.apimachinery.pkg.api.resource import (
+    generated_pb2 as api_resource,
 )
-from armada_jupyter.submissions import Submission
+from armada_jupyter.armada import create_armada_request
+from armada_jupyter.submissions import Job
+
 
-channel = grpc.insecure_channel("test")
-client = ArmadaClient(channel)
+@pytest.fixture(scope="session", autouse=True)
+def server_mock():
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
+    server.add_insecure_port("[::]:50051")
```

Comment:
Can we use a factory that spits out monotonically increasing ports to avoid flakiness?

Reply:
What if we supported using and ENV var for the port?

Reply:
Yea I think that would be fine.  These tests aren't really flaky since they will run in standalone CI.  The only issue is if you try to run these tests while also running Armada on your dev machine.  

I usually change it to the non default port but a ENV var would be fine.

Reply:
Sorry - I just realised what was meant by this - I am not sure how we would reliablly set the port dynamically (Maybe with it increasing based on a time hash?), but I think setting it to a different constant random port makes sense.